library/pkparse.c

```diff
@@ -446,6 +446,154 @@ static int pk_group_id_from_specified(const mbedtls_asn1_buf *params,
+    mbedtls_ecp_group_id ecp_group_id;
+    int ret;
+
+    ecp_group_id = mbedtls_ecc_group_of_psa(pk->MBEDTLS_PRIVATE(pk_ec_family),
+                                            pk->MBEDTLS_PRIVATE(pk_bits), 0);
+
+    mbedtls_ecp_keypair_init(&ecp_key);
+    ret = mbedtls_ecp_group_load(&(ecp_key.grp), ecp_group_id);
+    if (ret < 0) {
+        return ret;
+    }
+    ret = mbedtls_ecp_point_read_binary(&(ecp_key.grp), &ecp_key.Q,
+                                        in_start, in_len);
+    if (ret < 0) {
+        mbedtls_ecp_keypair_free(&ecp_key);
+        return ret;
+    }
+    ret = mbedtls_ecp_point_write_binary(&(ecp_key.grp), &ecp_key.Q,
+                                         MBEDTLS_ECP_PF_UNCOMPRESSED,
+                                         out_buf_len, out_buf, out_buf_size);
+    if (ret < 0) {
+        mbedtls_ecp_keypair_free(&ecp_key);
+        return ret;
+    }
+    mbedtls_ecp_keypair_free(&ecp_key);
+
+    return 0;
+}
+
+static int pk_get_ecpubkey(unsigned char **p, const unsigned char *end,
+                           mbedtls_pk_context *pk)
+{
+    mbedtls_svc_key_id_t key;
+    psa_key_attributes_t key_attrs = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status;
+    size_t len = (end - *p);
+    int ret;
+
+    if ((len > PSA_EXPORT_PUBLIC_KEY_MAX_SIZE) ||
+        (*p == NULL) || (end == NULL) ||
+        (pk == NULL)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+
+    /* Compressed point format are not supported yet by PSA crypto. As a
+     * consequence ecp functions are used to "convert" the point to
+     * uncompressed format */
+    if ((**p == 0x02) || (**p == 0x03)) {
+        ret = pk_convert_compressed_ec(pk, *p, len,
+                                       &(pk->MBEDTLS_PRIVATE(pk_raw_len)),
+                                       pk->MBEDTLS_PRIVATE(pk_raw),
+                                       PSA_EXPORT_PUBLIC_KEY_MAX_SIZE);
+        if (ret < 0) {
+            return ret;
+        }
+    } else {
+        /* Uncompressed format */
+        memcpy(pk->MBEDTLS_PRIVATE(pk_raw), *p, (end - *p));
+        pk->MBEDTLS_PRIVATE(pk_raw_len) = end - *p;
+    }
+
+    /* Validate the key by trying to importing it */
+    psa_set_key_usage_flags(&key_attrs, 0);
+    psa_set_key_algorithm(&key_attrs, PSA_ALG_ECDSA_ANY);
+    psa_set_key_type(&key_attrs,
+                     PSA_KEY_TYPE_ECC_PUBLIC_KEY(pk->MBEDTLS_PRIVATE(pk_ec_family)));
+    psa_set_key_bits(&key_attrs, pk->MBEDTLS_PRIVATE(pk_bits));
+
+    status = psa_import_key(&key_attrs, pk->MBEDTLS_PRIVATE(pk_raw),
+                            pk->MBEDTLS_PRIVATE(pk_raw_len), &key);
+    psa_destroy_key(key);
+    if (status != PSA_SUCCESS) {
+        mbedtls_platform_zeroize(pk->MBEDTLS_PRIVATE(pk_raw),
+                                 MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN);
+        pk->MBEDTLS_PRIVATE(pk_raw_len) = 0;
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+
+    *p = (unsigned char *) end;
```

Comment:
```suggestion
    *p = end;
```

Reply:
Unfortunately this change cannot be applied because the otherwise the `const` qualifier would be discarded and the compiler complains